lib/web_ui/lib/src/engine/text/layout_service.dart

```diff
@@ -263,6 +263,57 @@ class TextLayoutService {
     }
     return boxes;
   }
+
+  ui.TextPosition getPositionForOffset(ui.Offset offset) {
+    // [offset] is above all the lines.
+    if (offset.dy < 0) {
+      return ui.TextPosition(offset: 0, affinity: ui.TextAffinity.downstream);
+    }
+
+    // [offset] is below all the lines.
+    if (offset.dy >= paragraph.height) {
+      return ui.TextPosition(
+        offset: paragraph.toPlainText().length,
+        affinity: ui.TextAffinity.upstream,
+      );
+    }
+
+    final EngineLineMetrics line = _findLineForY(offset.dy);
+    // [offset] is to the left of the line.
+    if (offset.dx <= line.left) {
+      return ui.TextPosition(
+        offset: line.startIndex,
+        affinity: ui.TextAffinity.downstream,
+      );
+    }
+
+    // [offset] is to the right of the line.
+    if (offset.dx >= line.left + line.widthWithTrailingSpaces) {
+      return ui.TextPosition(
+        offset: line.endIndexWithoutNewlines,
+        affinity: ui.TextAffinity.upstream,
+      );
+    }
+
+    final double dx = offset.dx - line.left;
+    for (final RangeBox box in line.boxes!) {
+      if (box.left <= dx && dx <= box.right) {
+        return box.getPositionForX(dx);
+      }
+    }
+    // Is this ever reachable?
+    return ui.TextPosition(offset: line.startIndex);
+  }
+
+  EngineLineMetrics _findLineForY(double y) {
```

Comment:
If you agree, add a comment: // Not worth using binary search with a line.yOffset since number of lines is typically low.  Or something to that effect

Reply:
Done.